src/NUnitFramework/framework/Internal/Execution/WorkShift.cs

```diff
@@ -132,12 +132,13 @@ public void Start()
             IsActive = true;
 
             if (_firstStart)
+            {
+                _firstStart = false;
```

Comment:
Is there any chance of a race between threads for this check? If so, we could do:

```cs
// _firstStart will need to be an int with 0 for false and 1 for true
if (Interlocked.Exchange(ref _firstStart, 0) == 1)
{
```

Reply:
From what I can tell, as long as the flag gets set before the background thread is kicked off, there should be no threading risk here.